packages/actor-extract-links-extract-tree/lib/treeMetadataExtraction.ts

```diff
@@ -0,0 +1,86 @@
+import type * as RDF from 'rdf-js';
+import { termToString } from 'rdf-string';
+import { TreeNodes, RelationOperatorReversed } from './TreeMetadata';
+import type { ITreeRelation, ITreeRelationRaw, SparqlRelationOperator } from './TreeMetadata';
+
+/**
+ * Materialize a raw tree relation using the captured values.
+ * @param relationRaw Raw representation of a tree relation.
+ * @param nextLink Link to the next page.
+ * @returns ITreeRelation
+ */
+export function materializeTreeRelation(
+  relationRaw: ITreeRelationRaw,
+  nextLink: string,
+): ITreeRelation {
+  const relation: ITreeRelation = { node: nextLink };
+  if (relationRaw?.operator) {
+    relation.type = relationRaw.operator[0];
+  }
+
+  if (relationRaw?.remainingItems) {
+    relation.remainingItems = relationRaw.remainingItems[0];
+  }
+
+  if (relationRaw?.subject) {
+    relation.path = relationRaw.subject[0];
+  }
+
+  if (relationRaw?.value) {
+    relation.value = {
+      value: relationRaw.value[0],
+      term: relationRaw.value[1].object,
+    };
+  }
+
+  return relation;
+}
+
+/**
+ * From a quad stream return a relation element if it exist
+ * @param {RDF.Quad} quad - Current quad of the stream.
+ * @returns {[SparqlRelationOperator | number | string, keyof ITreeRelationRaw] | undefined} The relation element
+ * and the key associated with it.
+ */
+export function buildRelationElement(
+  quad: RDF.Quad,
+): [SparqlRelationOperator | number | string, keyof ITreeRelationRaw] | undefined {
```

Comment:
I don't really understand what this function does.
What are the different return types of this function?
Do we perhaps need a dedicated interface with different implementations?